test/unit_test/automl_test.cc

```diff
@@ -10,16 +10,19 @@
 #include <boost/test/test_tools.hpp>
 
 #include "test_common.h"
-#include "metric_sink.h"
 #include "simulator.h"
 #include "reductions_fwd.h"
+#include "test_red.cc"
```

Comment:
that's.... nasty.

Reply:
A better approach would be to have a test_red_private.h that is used here and don't force us to include the implementation

Reply:
this is a draft btw, we're still focused on features and we're cutting corners in plenty of places so beware (see learner.h for more). There's a couple of items to revisit once we're feature/functional complete in terms of the mvp

Reply:
that is a good idea, I've seen dirs titled internal for those sort of internal headers that shouldn't be consumed for the public

Reply:
removed .cc include, resolving this convo.